### PR TITLE
Update immunization store for AB#14892

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#14892](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14892)

## Description

Update Immunization store so that immunizations are stored by hdid.

**Timeline**

Screenshot 2023-02-09 at 3 59 24 PM

**Immunization Timeline Entry Card:**

Screenshot 2023-02-09 at 4 00 06 PM

**Covid 19 Timeline Entry Card:**

Screenshot 2023-02-09 at 3 59 54 PM

**Covid 19 Card**

Screenshot 2023-02-09 at 3 59 32 PM

Screenshot 2023-02-09 at 3 59 39 PM

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by ytqsl — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/store/modules/immunization/util.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+export function getImmunizationDatasetState(
+    state: ImmunizationState,
+    hdid: string
+): ImmunizationDatasetState {
+    return state.immunizations[hdid] ?? { ...defaultImmunizationDatasetState };
+}
+
+/**
+ * Updates the health visit state for a particular HDID.
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/store/modules/immunization/util.ts:30]: Nit: "health visit" should be "immunization".

File: Apps/WebClient/src/ClientApp/src/store/modules/immunization/getters.ts [hunk 2]
```diff
@@ -4,47 +4,60 @@ import { ImmunizationEvent, Recommendation } from "@/models/immunizationModel";
...
-        return (
-            state.status === LoadStatus.DEFERRED ||
-            state.status === LoadStatus.ASYNC_REQUESTED
-        );
+    isDeferredLoad(state: ImmunizationState): (hdid: string) => boolean {
+        return (hdid: string) =>
+            getImmunizationDatasetState(state, hdid).status ===
+                LoadStatus.DEFERRED ||
+            getImmunizationDatasetState(state, hdid).status ===
+                LoadStatus.ASYNC_REQUESTED;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/store/modules/immunization/getters.ts:56]: A small optimization would be to retrieve the state just once:

```
return (hdid: string) => {
    const datasetState = getImmunizationDatasetState(state, hdid);
    return (
        datasetState.status === LoadStatus.DEFERRED ||
        datasetState.status === LoadStatus.ASYNC_REQUESTED
    );
};
````

File: Apps/WebClient/src/ClientApp/src/store/modules/immunization/actions.ts [hunk 3]
```diff
@@ -18,12 +18,15 @@ export const actions: ImmunizationActions = {
         );
 
         return new Promise((resolve, reject) => {
-            if (context.state.status === LoadStatus.LOADED) {
+            if (
+                context.state.immunizations[params.hdid]?.status ===
+                LoadStatus.LOADED
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/store/modules/immunization/actions.ts:23]: I suggest swapping to use `getImmunizationDatasetState(context.state, params.hdid)` as it is slightly safer in general and that's what we're doing in the other modules now.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/store/modules/immunization/util.ts (hunk 1 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/store/modules/immunization/util.ts]: Fixed

File: Apps/WebClient/src/ClientApp/src/store/modules/immunization/actions.ts (hunk 3 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/store/modules/immunization/actions.ts:23]: Fixed.

File: Apps/WebClient/src/ClientApp/src/store/modules/immunization/getters.ts (hunk 2 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/store/modules/immunization/getters.ts]: Fixed

File: Apps/WebClient/src/ClientApp/src/store/modules/immunization/actions.ts (hunk 3 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/store/modules/immunization/actions.ts]: Fixed

**Pull request merged**